### PR TITLE
Fixes #5160. Rundeck to load specified config file

### LINKED ISSUE #5160: Rundeck ignores custom config file name
**Describe the bug**

In my deployment I have a `rundeck-config_WITH_SECRETS.groovy` file but I cannot configure Rundeck to load it. 

Rundeck ignores the value of `-Drundeck.config.location` and `-Drundeck.config.file` attempting to always load `rundeck-config.groovy`

This is the line that is causing it: https://github.com/rundeck/rundeck/blob/978c0ac046c8daee408348fcd96a29a8c64ef3cc/rundeckapp/grails-app/init/rundeckapp/Application.groovy#L90

**My Rundeck detail**
* Rundeck version: 3.X

**Expected behavior**

Rundeck should take into account environment variables values (`${RDECK_CONFIG_FILE}` and `${RDECK_CONFIG_LOCATION}`)

### PR DESCRIPTION
**Pull request by danifr**
**Is this a bugfix, or an enhancement? Please describe.**
Bugfix for: https://github.com/rundeck/rundeck/issues/5160

**Describe the solution you've implemented**
Rundeck should get the proper config file

### CODE REVIEW

**Comment by danifr** [comment]
Not sure why the CI is not passing though. Maybe @ProTip can help me out here. Thanks!

**Comment by sjrd218** [comment]
@danfir Thanks for the PR. Can you add a test that shows your new code working with either `RundeckInitConfig.SYS_PROP_RUNDECK_CONFIG_LOCATION` as you have it set, or the default which is to use the default config dir plus the `rundeck-config.groovy`. Some people just dump a `rundeck-config.groovy` in the config dir and expect it to be picked up.

**Comment by danifr** [comment]
Sure thing! But I'm not sure if I follow you. Can you assist me in how should I create the test?

The default value (from `/etc/rundeck/profile file) that we can find in a new installation of Rundeck is:
```
RDECK_CONFIG_FILE="${RDECK_CONFIG_FILE:-$RDECK_CONFIG/rundeck-config.properties}"
```
People wanting to load a different file would need to override that variable.

I guess `RDECK_CONFIG_LOCATION` and `RDECK_CONFIG_FILE` should point to the same file.
https://github.com/rundeck/rundeck/blob/978c0ac046c8daee408348fcd96a29a8c64ef3cc/rundeckapp/grails-app/init/rundeckapp/Application.groovy#L64

**Comment by sjrd218** [comment]
I was thinking just adding a unit test to: https://github.com/rundeck/rundeck/blob/master/rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy that check for both conditions would be great.

**Comment by danifr** [comment]
Not sure if this is what you had in mind for the tests. Let me know if I need to modify the tests

File: rundeckapp/grails-app/init/rundeckapp/Application.groovy [hunk 1]
```diff
@@ -86,8 +86,7 @@ class Application extends GrailsAutoConfiguration implements EnvironmentAware {
     }
 
     void loadGroovyRundeckConfigIfExists(final Environment environment) {
-        String rundeckGroovyConfigFile = System.getProperty(RundeckInitConfig.SYS_PROP_RUNDECK_SERVER_CONFIG_DIR) +
-                                         "/rundeck-config.groovy"
+        String rundeckGroovyConfigFile = System.getProperty(RundeckInitConfig.SYS_PROP_RUNDECK_CONFIG_LOCATION)
```
Comment (sjrd218) [inline rundeckapp/grails-app/init/rundeckapp/Application.groovy]: Your code should check if there is a file provided by RundeckInitConfig.SYS_PROP_RUNDECK_CONFIG_LOCATION, and if not, it needs to check it there is a rundeck-config.groovy in the server config directory, which is what the previous code would load.

File: rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy [hunk 2]
```diff
@@ -68,6 +68,40 @@ class ApplicationTest extends Specification {
 
     }
 
+    def "load rundeck-config.groovy if set in system property RDECK_CONFIG_LOCATION"() {
+        when:
+        File tmpGroovy = File.createTempFile("rundeck-config",".groovy")
+        tmpGroovy << "grails { mail {} } "
+        System.setProperty(RundeckInitConfig.SYS_PROP_RUNDECK_CONFIG_LOCATION,tmpGroovy.absolutePath)
```
Comment (sjrd218) [inline rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy]: This test should check if RundeckInitConfig.SYS_PROP_RUNDECK_CONFIG_LOCATION is not set, but there is a rundeck-config.groovy file in the system config directory. If there is a file there, we want to pick it up and load it.

**Comment by danifr** [comment]
HI @sjrd218 , thanks for the feedback. I think I addressed your comments in my latest commmit. Let me know

File: rundeckapp/grails-app/init/rundeckapp/Application.groovy (hunk 1 above)
Comment (danifr) [reply rundeckapp/grails-app/init/rundeckapp/Application.groovy]: https://github.com/danifr/rundeck/blob/28f838350d228f9750aff8048ff366e04e18c65f/rundeckapp/grails-app/init/rundeckapp/Application.groovy#L92

File: rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy (hunk 2 above)
Comment (danifr) [reply rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy]: https://github.com/danifr/rundeck/blob/28f838350d228f9750aff8048ff366e04e18c65f/rundeckapp/src/test/groovy/rundeckapp/ApplicationTest.groovy#L73

**Comment by sjrd218** [comment]
@danfir Very nice! Thanks for your contribution.

**Pull request merged**